tests/unit/test_caiso.py

```diff
@@ -200,3 +194,76 @@ def test_bad_data_lmp_only(self, exptected_request):
+        self.assertEqual(generation[239]['timestamp'], Timestamp('2017-11-05T06:00:00Z'))  # '2017-11-04T23:00:00-07:00'
+        self.assertEqual(generation[239]['fuel_name'], 'hydro')
+        self.assertAlmostEqual(generation[239]['gen_MW'], 2426)
+
+    @requests_mock.Mocker()
+    def test_get_generation_dst_end(self, mock_request):
+        expected_url = 'http://content.caiso.com/green/renewrpt/20171105_DailyRenewablesWatch.txt'
+        expected_response = read_fixture(self.c.__module__, '20171105_DailyRenewablesWatch.txt').encode('utf-8')
+        mock_request.get(expected_url, content=expected_response)
+
+        start_at = parse('2017-11-05T00:00:00-07:00')
+        end_at = parse('2017-11-05T23:59:59-08:00')
+        generation = self.c.get_generation(start_at=start_at, end_at=end_at,
+                                           # FIXME: Non-base kwargs are required to route to _generation_historical()
+                                           market=self.c.MARKET_CHOICES.hourly, freq=self.c.FREQUENCY_CHOICES.hourly)
+
+        self.assertEqual(generation[0]['timestamp'], Timestamp('2017-11-05T07:00:00Z'))  # '2017-11-05T00:00:00-07:00'
+        self.assertEqual(generation[9]['timestamp'], Timestamp('2017-11-05T08:00:00Z'))  # '2017-11-05T01:00:00-07:00'
+        # "Hour 1" appears only once in 25 hour day. The 01:00-08:00 hour is skipped, leaving a gap in UTC timeseries.
+        self.assertEqual(generation[18]['timestamp'], Timestamp('2017-11-05T010:00:00Z'))  # '2017-11-05T02:00:00-08:00'
+        self.assertEqual(generation[239]['timestamp'], Timestamp('2017-11-06T07:00:00Z'))  # '2017-11-05T23:00:00-08:00'
+
+    @requests_mock.Mocker()
+    def test_get_generation_standard_time(self, mock_request):
+        expected_url = 'http://content.caiso.com/green/renewrpt/20171106_DailyRenewablesWatch.txt'
+        expected_response = read_fixture(self.c.__module__, '20171106_DailyRenewablesWatch.txt').encode('utf-8')
+        mock_request.get(expected_url, content=expected_response)
+
+        start_at = parse('2017-11-06T00:00:00-08:00')
+        end_at = parse('2017-11-06T23:59:59-08:00')
+        generation = self.c.get_generation(start_at=start_at, end_at=end_at,
+                                           # FIXME: Non-base kwargs are required to route to _generation_historical()
+                                           market=self.c.MARKET_CHOICES.hourly, freq=self.c.FREQUENCY_CHOICES.hourly)
+
+        self.assertEqual(generation[0]['timestamp'], Timestamp('2017-11-06T08:00:00Z'))  # '2017-11-06T00:00:00-08:00'
+        self.assertEqual(generation[0]['fuel_name'], 'solarth')
+        self.assertAlmostEqual(generation[0]['gen_MW'], 0)
+        self.assertEqual(generation[239]['timestamp'], Timestamp('2017-11-07T07:00:00Z'))  # '2017-11-06T23:00:00-08:00'
+        self.assertEqual(generation[239]['fuel_name'], 'hydro')
+        self.assertAlmostEqual(generation[239]['gen_MW'], 1969)
```

Comment:
Lines 197-269 add four unit tests with mocked requests:
- typical standard time day
- transition day into daylight saving time
- typical daylight saving time day
- transition day back to standard time